### PR TITLE
expect/actual removing abstract modifier

### LINKED ISSUE #1001: expect/actual removing abstract modifier
## Describe the bug

expect/actual removing abstract modifier 

## Expected behavior

actual abstract class CoroutineTest actual constructor() {

    /**
     * Test rule
     */
    @get:Rule
    var coroutineTestRule = CoroutineTestRule()

    /**
     * Run test
     *
     * @param T
     * @param block
     * @receiver a Coroutine Scope
     */
    actual fun <T> runTest(block: suspend CoroutineScope.() -> T) {
        runBlocking {
            block()
        }
    }
}

## Observed behavior

@RunWith(AndroidJUnit4::class)
actual class CoroutineTest actual constructor() {
    /**
     * Test rule
     */
    @get:Rule
    var coroutineTestRule = CoroutineTestRule()

    /**
     * Run test
     *
     * @param T
     * @param block
     * @receiver a Coroutine Scope
     */
    actual fun <T> runTest(block: suspend CoroutineScope.() -> T) {
        runBlocking {
            block()
        }
    }
}

## Environment information
* diktat version: 1.0.0-rc.2
* build tool (maven/gradle): gradle
* how is diktat run (CLI, plugin, etc.): plugin
* kotlin version: 1.5.21
* operating system: mac
* link to a project (if your project is public): N/A

### PR DESCRIPTION
**Pull request by Cheshiriks**
### What's done:
* fixed bug in CLASS_SHOULD_NOT_BE_ABSTRACT
* now replace abstract modifier on open modifier
Closes #1001

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/AbstractClassesRule.kt [hunk 1]
```diff
@@ -45,25 +45,9 @@ class AbstractClassesRule(configRules: List<RulesConfig>) : DiktatRule(
...
-                        modList.removeChild(spaceInModifiers)
-                    }
-                } else {
-                    if (modList.treeNext.isWhiteSpace()) {
-                        classNode.removeChild(modList.treeNext)
-                    }
-                    classNode.removeChild(modList)
-                }
+                val abstractKeyword = modList.getFirstChildWithType(ABSTRACT_KEYWORD)!!
+                val newOpenKeyword = KotlinParser().createNode("open")
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/AbstractClassesRule.kt:49]: This change seems simple enough not to invoke KotlinParser. Can we just replace node with `asbstract` with a new `LeafPsiElement`?

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter6/AbstractClassesWarnTest.kt [hunk 2]
```diff
@@ -15,7 +15,7 @@ class AbstractClassesWarnTest : LintTestBase(::AbstractClassesRule) {
 
     @Test
     @Tag(CLASS_SHOULD_NOT_BE_ABSTRACT)
-    fun `should not remove abstract`() {
+    fun `should not replace abstract on open`() {
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter6/AbstractClassesWarnTest.kt:18]:
```suggestion
    fun `should not replace abstract with open`() {
```

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/AbstractClassesRule.kt (hunk 1 above)
Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/AbstractClassesRule.kt]: Yes. Fixed

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/AbstractClassesRule.kt [hunk 3]
```diff
@@ -46,8 +46,7 @@ class AbstractClassesRule(configRules: List<RulesConfig>) : DiktatRule(
             CLASS_SHOULD_NOT_BE_ABSTRACT.warnAndFix(configRules, emitWarn, isFixMode, identifier, node.startOffset, node) {
                 val modList = classNode.getFirstChildWithType(MODIFIER_LIST)!!
                 val abstractKeyword = modList.getFirstChildWithType(ABSTRACT_KEYWORD)!!
-                val newOpenKeyword = KotlinParser().createNode("open")
-                modList.replaceChild(abstractKeyword, newOpenKeyword)
+                (abstractKeyword as LeafPsiElement).rawReplaceWithText("open")
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/AbstractClassesRule.kt:49]: This won't change node type, will it?

Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/AbstractClassesRule.kt:49]: Yes. Created new LeafPsiElement

**Review by petertrr — APPROVED** [review]

**Review by petertrr — CHANGES_REQUESTED** [review]
Please also update docs

**Review by petertrr — APPROVED** [review]

**Pull request merged**